src/sentry/static/sentry/app/components/buttons/button.jsx

```diff
@@ -90,70 +74,170 @@ class Button extends React.Component {
       ...buttonProps
     } = this.props;
 
-    let isPrimary = priority === 'primary' && !disabled;
-    let isDanger = priority === 'danger' && !disabled;
-    let isSuccess = priority === 'success' && !disabled;
-    let isLink = priority === 'link';
-
-    let cx = classNames(className, 'button', {
+    let cxs = classNames(className, {
       tip: !!title,
-      'button-no-border': borderless,
-      'button-primary': isPrimary,
-      'button-danger': isDanger,
-      'button-success': isSuccess,
-      'button-link': isLink && !isPrimary && !isDanger,
-      'button-default': !isLink && !isPrimary && !isDanger,
-      'button-zero': size === 'zero',
-      'button-sm': size === 'small',
-      'button-xs': size === 'xsmall',
-      'button-lg': size === 'large',
-      'button-busy': busy,
-      'button-disabled': disabled,
     });
 
-    let childContainer = (
-      <Flex align="center" className="button-label">
-        {icon && (
-          <Icon size={size}>
-            <StyledInlineSvg src={icon} size={size === 'small' ? '12px' : '16px'} />
-          </Icon>
-        )}
-        {children}
-      </Flex>
-    );
-
     // Buttons come in 3 flavors: Link, anchor, and regular buttons. Let's
     // use props to determine which to serve up, so we don't have to think
     // about it. As a bonus, let's ensure all buttons appear as a button
     // control to screen readers. Note: you must still handle tabindex manually.
 
-    // Props common to all elements
-    let componentProps = {
-      disabled,
-      ...buttonProps,
-      onClick: this.handleClick,
-      className: cx,
-      role: 'button',
-      children: childContainer,
-    };
-
-    // Handle react-router Links
-    if (to) {
-      return <Link to={this.getUrl()} {...componentProps} />;
-    }
+    return (
+      <StyledSmartButton
```

Comment:
Could use a better name here

Reply:
it isn't *always* a styled import, so there is even debate wether we should use the "styled" prefix. I would vote no since it's basically a component that *might* return an imported component. 

As for the name: `ButtonOrLinkElement` is direct. `SmartHtmlElement` is another option i'd be ok with. If you really wanted to get obvious there is always `HtmlElementBasedOnProps` lol.

Reply:
It is always a styled component though. The component we are styling is `ButtonOrLinkElement`. Thinking about it, maybe it's better to not expose this through the component name and just name it `StyledButton`.